client/src/document/ingredients/browser-compatibility-table/utils.ts

```diff
@@ -70,3 +70,16 @@ export function hasNoteworthyNotes(support: bcd.SimpleSupportStatement) {
     !support.partial_implementation
   );
 }
+
+export function needsPrefix(support: bcd.SupportStatement | undefined) {
+  return (
+    support &&
+    getFirst(support).prefix &&
+    !asList(support).some(
+      (item) =>
+        Object.keys(item).filter(
+          (x) => !["version_added", "release_date"].includes(x)
+        ).length === 0
```

Comment:
This might break if we add an additional key to `bcd.SupportStatement`, so let's be explicit:

```suggestion
        item.version_added
        && !item.version_removed
        && !item.partial_implementation
        && !item.alternative_name
        && !item.flags
        && !item.prefix
```

Reply:
Ok, should I also change https://github.com/mdn/yari/blob/main/client/src/document/ingredients/browser-compatibility-table/feature-row.tsx#L386-L391? That's where I got this check

Reply:
This check was added in https://github.com/mdn/yari/pull/3864

Reply:
Good point! Can you tell me what you think those lines (389-392 and 398-400) do? Something like `hasFullSupport(item)` and `hasNoSupport(item)`?

This is exactly why I prefer self-documenting functions (and you add one in this PR, so that's great!). 😄

Reply:
I don't know the exact best phrasing for the internal function, but I'm thinking something like:
```
function hasSpecialProperty(support: bcd.SimpleSupportStatement) {
  return (
    support.version_removed ||
    support.partial_implementation ||
    support.alternative_name ||
    support.flags ||
    support.prefix
  );
}

export function hasFullSupport(support: bcd.SimpleSupportStatement) {
  return (
    support.version_added && !hasSpecialProperty(support)
  );
}

export function hasNoSupport(support: bcd.SimpleSupportStatement) {
  return (
    !support.version_added && !hasSpecialProperty(support)
  );
}
```

Reply:
I seem to be unable to push these changes right now: https://www.githubstatus.com/

Reply:
Instead of `hasSpecialProperty()`, maybe `has[Some]Limitation()` or `has[Some]Restriction()`?

And I have to admit that the existing `hasNoSupport()` logic is not really intuitive to me. If we have `!support.version_added` but `support.version_removed`, doesn't this also mean "No support"? 🤔

Reply:
Can you think of anything that you've seen with a version removed but no version added? I can't find anything like that. Maybe this just works because such an item doesn't exist.

Reply:
Actually, I just realized what this check is really doing. It's not just checking if there's no support. It's checking if there's no support and no other messaging is going to appear. It's not what determines if the symbol shows (that's in CellText), it's what determines if the line "No support" shows.

Reply:
![Screenshot from 2022-03-22 14-34-21](https://user-images.githubusercontent.com/29206584/159551894-f68f5563-ffec-44e8-8ef9-9f71dd0515b6.png)